servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ExpectContinueTest.java

```diff
@@ -0,0 +1,251 @@
+            payload.awaitSubscribed();
+            payload.onNext(client.executionContext().bufferAllocator().fromAscii(PAYLOAD));
+            payload.onComplete();
+
+            assertResponse(OK, PAYLOAD);
+        }
+    }
+
+    @ParameterizedTest(name = "protocol={0}")
+    @EnumSource(HttpProtocol.class)
+    void serverRespondsWithSuccess(HttpProtocol protocol) throws Exception {
+        try (HttpServerContext serverContext = startServer(protocol, (ctx, request, response) -> {
+                requestReceived.countDown();
+                returnResponse.await();
+                response.status(ACCEPTED);
+                try (HttpPayloadWriter<Buffer> writer = response.sendMetaData()) {
+                    for (Buffer chunk : request.payloadBody()) {
+                        writer.write(chunk);
+                    }
+                }
+            });
+             StreamingHttpClient client = createClient(serverContext, protocol)) {
+
+            TestPublisher<Buffer> payload = new TestPublisher.Builder<Buffer>().singleSubscriber().build();
+            client.request(client.post("/").setHeader(EXPECT, CONTINUE).payloadBody(payload))
+                    .subscribe(responses::add);
+            requestReceived.await();
+
+            assertThat("Unexpected subscribe to payload body before 100 (Continue)",
+                    payload.isSubscribed(), is(false));
+            returnResponse.countDown();
+
+            StreamingHttpResponse response = responses.take();
+            assertThat(response.status(), is(ACCEPTED));
+
+            payload.awaitSubscribed();
+            payload.onNext(client.executionContext().bufferAllocator().fromAscii(PAYLOAD));
+            payload.onNext(client.executionContext().bufferAllocator().fromAscii(PAYLOAD));
+            payload.onComplete();
+
+            assertThat(response.toResponse().toFuture().get().payloadBody().toString(US_ASCII),
+                    equalTo(PAYLOAD + PAYLOAD));
+        }
+    }
+
+    @ParameterizedTest(name = "protocol={0}")
+    @EnumSource(HttpProtocol.class)
+    void expectationFailed(HttpProtocol protocol) throws Exception {
+        try (HttpServerContext serverContext = startServer(protocol);
+             StreamingHttpClient client = createClient(serverContext, protocol)) {
+
+            TestPublisher<Buffer> payload = new TestPublisher.Builder<Buffer>().singleSubscriber().build();
+            client.request(client.post("/").setHeader(EXPECT, CONTINUE).payloadBody(payload))
+                    .subscribe(responses::add);
+            requestReceived.await();
+
+            assertThat("Unexpected subscribe to payload body before 100 (Continue)",
+                    payload.isSubscribed(), is(false));
+            returnResponse.countDown();
+
+            // FIXME: response currently blocks because it waits for request payload body write to finish,
+            //  cancel request payload body write (WSS#cancel/discard)
```

Comment:
@Scottmitch this is a draft PR that just puts all required pieces together to make tests work, but I still struggle to find what needs to be done on the client-side when the server responds with an error to unblock response completion. Would be happy to chat with you